src/client.coffee

```diff
@@ -238,6 +238,45 @@ module.exports = class Client
     this._do_request_response req_opts, callback, "Client::authrep_with_user_key"
 
 
+
+  ###
+    Authorize and Report in a single call with OAuth
+    ---------------------------------------------------
+    
+    Parameters:
+      options is a Hash object with the following fields
+        service_token Required if you didn't use provider_key to ceate the Client instance
+        user_key Required
+        service_id Required (from November 2016)
+      callback {Function} Is the callback function that receives the Response object which includes `is_success` method to determine the status of the response
+
+    Example using provider_key:
+      client.authrep_with_user_key {service_id: '1234567890987', user_key: 'ca5c5a49'}, (response) ->
+        if response.is_success
+          # All Ok
+        else
+         sys.puts "#{response.error_message} with code: #{response.error_code}"
+  
+    Example using service_token:
+      client.authrep_with_user_key {service_token: '12sdtsdr23454sdfsdf', service_id: '1234567890987', user_key: 'ca5c5a49'}, (response) ->
+        if response.is_success
+          # All Ok
+        else
+         sys.puts "#{response.error_message} with code: #{response.error_code}"
+  ###
+
+  authrep_with_oauth: (options, callback) ->
```

Comment:
`oauth_authrep` would probably be more consistent naming with all the other functions.